docs/recent-releases/go-ipfs-0-5/features.md

```diff
@@ -0,0 +1,160 @@
+
+1. When connected to the public internet, IPFS will use both DHTs for finding peers, content, and IPNS records. Nodes only publish provider and IPNS records to the WAN DHT to avoid flooding the local network.
+2. When not connected to the public internet, nodes publish provider and IPNS records to the LAN DHT.
+
+The WAN DHT includes all peers with at least one public IP address. This release will only consider an IPv6 address public if it is in the [public internet range `2000::/3`](https://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml).
+
+This feature should not have any noticeable impact on Go-IPFS, performance, or otherwise. Everything should continue to work in all the currently supported network configurations: VPNs, disconnected LANs, public internet, etc.
+
+#### Query Logic
+
+We've fixed the DHT query logic by correctly implementing Kademlia. This should significantly speed up:
+
+- Publishing IPNS & provider records. We previously continued searching for closer and closer peers to the "target" until we timed out, then we put to the closest peers we found.
+- Resolving IPNS addresses. We previously continued IPNS record searches until we ran out of peers to query, timed out, or found 16 records.
+
+In both cases, nodes continue searching until they find the closest peers and then stop.
+
+#### Routing Tables
+
+Finally, we've addressed the poorly maintained routing tables by:
+
+- Reducing the likelihood that the connection manager will kill connections to peers in the routing table.
+- Keeping peers in the routing table, even if we get disconnected from them.
+- Actively and frequently querying the DHT to keep our routing table full.
+
+### Testing
+
+The DHT rewrite was made possible by [Testground](https://github.com/ipfs/testground/), our new testing framework. Testground allows us to spin up multi-thousand node tests with simulated real-world network conditions. By combining Testground and some custom analysis tools, we were able to gain confidence that the new DHT implementation behaves correctly.
+
+## AutoNAT
+
+This release includes a protocol called AutoNAT for detecting whether or not a node is _reachable_ from the public internet. In short:
+
+1. An AutoNAT client asks a node running an AutoNAT service if it can be reached at one of a set of guessed addresses.
+2. The AutoNAT service attempts to _dial back_ those addresses, with some restrictions. We won't dial back to a different IP address, for example.
+3. If the AutoNAT service succeeds, it reports back the address it successfully dialed, and the AutoNAT client knows that it is reachable from the public internet.
+
+All nodes act as AutoNAT clients to determine if they should switch into DHT server mode. As of this release, all nodes also run a rate-limited AutoNAT service by default. This service should have minimal overhead. Nodes initialized with the `lowpower` configuration profile do not run the AutoNAT service.
```

Comment:
nodes will turn off their autonat service once they they feel confident they are not reachable by others.

"run a rate-limited AutoNAT service by default" can maybe be less scary as something to the effect of
```
As of this release, nodes will by default run the service side of AutoNAT - verifying connectivity - for up to 30 peers every minute. This service should have minimal overhead, and will be disabled for nodes in the `lowpower` configuration profile, and those which believe they are not publicly reachable.
```

Reply:
Addressed in 73eaab8 and 618d1ba.